modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java

```diff
@@ -1817,10 +1817,16 @@ public Schema getSchema(ObjectNode node, String location, ParseResult result){
             schema = oneOfList;
         } else if(itemsNode != null) {
             ArraySchema items = new ArraySchema();
-            for(JsonNode n : itemsNode) {
-                if(n.isValueNode()) {
-                    items.setItems(getSchema(itemsNode, location, result));
+            if (itemsNode.getNodeType().equals(JsonNodeType.OBJECT)){
+                items.setItems(getSchema(itemsNode, location, result));
+            }else if (itemsNode.getNodeType().equals(JsonNodeType.ARRAY)){
```

Comment:
which case is this handling? should it be perhaps a recursive call instead?

Reply:
hi I don't understand the question, getSchema() is a recursive call